### PR TITLE
support category translation

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/62018

This PR tackles the broken FE links for categories when WPML is active.

Note:

* WPML does not currently support translation of a custom tax base (e.g. `category`)
* we do support rewrites in our rewrite rules
* I've updated our code to support the category translation in our code when the user chooses to translate event categories (even though in WPML terms that would apply to *terms only*)
* I've also made it so that the non accented version of `category` translation is used by default (filterable, see code)
* I've also updated the rewrite rules handling code to add the non accented version of the category base to rewrite rules (for SEO purposes "catégorie" is not cool)

### CODE REVIEW

File: src/Tribe/Integrations/WPML/Category_Translation.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	 */
+	public function filter_tribe_events_category_slug( $slug ) {
+		/** @var SitePress $sitepress */
+		global $sitepress;
+
+		$tax_sync_options     = $sitepress->get_setting( 'taxonomies_sync_option' );
+		$translate_event_cat  = ! empty( $tax_sync_options[ Tribe__Events__Main::TAXONOMY ] );
+		$using_lang_query_var = $sitepress->get_setting( 'language_negotiation_type' ) == 3;
+
+		if ( !$translate_event_cat || $using_lang_query_var ) {
```
Comment (barryhughes) [inline src/Tribe/Integrations/WPML/Category_Translation.php]: Space after `!`

File: src/Tribe/Integrations/WPML/Category_Translation.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			$slug = 'category';
+		} else {
+			$lang        = $sitepress->get_locale( ICL_LANGUAGE_CODE );
```
Comment (barryhughes) [inline src/Tribe/Integrations/WPML/Category_Translation.php]: We could either remove the extra whitespace before `=` or else align the following lines.

File: src/Tribe/Integrations/WPML/Filters.php [hunk 3]
```diff
@@ -68,26 +68,40 @@ public function filter_tribe_events_rewrite_i18n_slugs_raw( $bases, $method, $do
 		$post_slug_translation_on  = ! empty( $sitepress_settings['posts_slug_translation']['on'] );
 
 		if ( $string_translation_active && $post_slug_translation_on ) {
-			$supported_post_types = array( Tribe__Events__Main::POSTTYPE );
+			$bases  = $this->translate_single_slugs( $bases );
```
Comment (barryhughes) [inline src/Tribe/Integrations/WPML/Filters.php]: Remove the extra space before `=`

File: src/Tribe/Integrations/WPML/Filters.php [hunk 4]
```diff
@@ -68,26 +68,40 @@ public function filter_tribe_events_rewrite_i18n_slugs_raw( $bases, $method, $do
...
+			$cpt_slug_is_not_translated = empty( $sitepress_settings['posts_slug_translation']['types'][ $post_type ] );
 
-				$bases['single'] = array_merge( $bases['single'], wp_list_pluck( $slug_translations[1], 'value' ) );
+			if ( $cpt_slug_is_not_translated ) {
+				continue;
 			}
+
+			$slug_translations = WPML_Slug_Translation::get_translations( $post_type );
+
+			if ( ! ( is_array( $slug_translations ) && isset( $slug_translations[1] ) ) ) {
```
Comment (barryhughes) [inline src/Tribe/Integrations/WPML/Filters.php]: Wondering, could this be simplified to just `if ( ! isset( $slug_translations[ 1 ] ) ) {}`?

File: src/Tribe/Integrations/WPML/WPML.php [hunk 5]
```diff
@@ -66,6 +66,11 @@ protected function hook_filters() {
 
 		$permalinks = Tribe__Events__Integrations__WPML__Permalinks::instance();
 		add_filter( 'post_type_link', array( $permalinks, 'filter_post_type_link' ), 20, 2 );
+
+		if ( !is_admin() ) {
```
Comment (barryhughes) [inline src/Tribe/Integrations/WPML/WPML.php]: Space after `!`

**Comment by lucatume** [comment]
fixes in place

**Pull request merged**